src/relation/builder.rs

```diff
@@ -246,6 +246,47 @@ impl<RequireInput> MapBuilder<RequireInput> {
         builder
     }
 
+    /// Initialize a builder with an existing map and filter by an `Expr` that depends on the input columns
+    pub fn filter_field_with(self, map: Map, predicate: Expr) -> MapBuilder<WithInput> {
```

Comment:
Why is this called `filter_field`? you are filtering all fields right? It is a bit misleading.
Let's change this to `filter_with`. As a general naming principle, `filter` implicitly filter rows of data and `filter_fields` filter columns.